google/cloud/aiplatform/models.py

```diff
@@ -1344,67 +1870,164 @@ def list(
-            self._gca_resource.traffic_split.keys(),
-            key=lambda id: self._gca_resource.traffic_split[id],
+        if len(self._gca_resource.deployed_models):
+            raise ValueError(
+                "A maximum of one model can be deployed to each private Endpoint. "
+                "Please call PrivateEndpoint.undeploy_all() before deploying another "
+                "model."
+            )
+
+        self._validate_deploy_args(
+            min_replica_count,
+            max_replica_count,
+            accelerator_type,
+            deployed_model_display_name,
+            explanation_metadata,
+            explanation_parameters,
         )
 
-        for deployed_model in models_to_undeploy:
-            self._undeploy(deployed_model_id=deployed_model, sync=sync)
+        self._deploy(
+            model=model,
+            deployed_model_display_name=deployed_model_display_name,
+            machine_type=machine_type,
+            min_replica_count=min_replica_count,
+            max_replica_count=max_replica_count,
+            accelerator_type=accelerator_type,
+            accelerator_count=accelerator_count,
+            service_account=service_account,
+            explanation_metadata=explanation_metadata,
+            explanation_parameters=explanation_parameters,
+            metadata=metadata,
+            sync=sync,
+        )
 
-        return self
+    def undeploy(
+        self,
+        deployed_model_id: str,
+        sync=True,
+    ) -> None:
+        """Undeploys a deployed model from the private Endpoint.
 
-    def delete(self, force: bool = False, sync: bool = True) -> None:
-        """Deletes this Vertex AI Endpoint resource. If force is set to True,
-        all models on this Endpoint will be undeployed prior to deletion.
+        Example Usage:
+
+            my_private_endpoint.undeploy(
+                deployed_model_id="1234567891232567891"
+            )
+
+            or
+
+            my_deployed_model_id = my_private_endpoint.list_models()[0].id
+            my_private_endpoint.undeploy(
+                deployed_model_id=my_deployed_model_id
+            )
 
         Args:
-            force (bool):
-                Required. If force is set to True, all deployed models on this
-                Endpoint will be undeployed first. Default is False.
-            sync (bool):
-                Whether to execute this method synchronously. If False, this method
-                will be executed in concurrent Future and any downstream object will
-                be immediately returned and synced when the Future has completed.
-        Raises:
-            FailedPrecondition: If models are deployed on this Endpoint and force = False.
+            deployed_model_id (str):
+                Required. The ID of the DeployedModel to be undeployed from the
+                private Endpoint. Use PrivateEndpoint.list_models() to get the
+                deployed model ID.
```

Comment:
It seems like `sync` was removed in the docstring. Hard to tell with this diff.

Reply:
Fixed